### DISCUSSION TITLE
How to set and update database on a node?

### DISCUSSION THREAD

**Discussion by rhyswynn** [Holmeswww/AgentKit#14] [Q&A] [answered]
I'm looking through all the documentation, and it seems like there's no way to 'set' the node.db database values in the existing code other than by creating the SimpleDBNode dynamically. Even that would need a custom function in a new object based on BaseAfterQuery. Am I missing something?

**Comment by Holmeswww** [comment, marked as answer]
Hi,

As shown [here](https://agentkit.readthedocs.io/en/latest/node%20in%20agentkit.html#agentkit.after_query.BaseAfterQuery), you can access `self.node` from after_query. This will give you full access to self.node.db.

Please let me know if this resolves your question

**Comment by rhyswynn** [comment]
Yes, that's where I thought it should go, but it means the code on the site 'as is' does not include an implementation of a function that sets or updates the db value. Perhaps this is more of the novelty? :)
